tools/tuf/releaser.sh

```diff
@@ -173,6 +164,21 @@ release_fleetd_to_edge () {
     popd
 }
 
+create_fleetd_release_pr () {
+    echo "Creating a PR for fleetd release changelog..."
+    BRANCH_NAME=release-fleetd-v$VERSION
+    pushd "$GIT_REPOSITORY_DIRECTORY"
+    git checkout -b "$BRANCH_NAME"
+    make changelog-orbit version="$VERSION"
+    ORBIT_CHANGELOG=orbit/CHANGELOG.md
+    git add "$ORBIT_CHANGELOG"
+    git commit -m "Release fleetd $VERSION"
+    git push origin "$BRANCH_NAME"
+    # Create a new PR with the changelog.
+    gh pr create -f -B main -t "Release fleetd $VERSION"
```

Comment:
Double checking: Will there be issues if `main` and whatever `rc-minor-fleet-v4.XX.YY` have very different commit histories? (we are only interested on merging changes under `orbit/changes/` and `CHANGELOG.md`).

Reply:
Ah yes -- that was my question as well, glad you resurfaced this.  I think for patch releases we really want to be doing a PR against the branch that was originally checked out, rather than main.  So what I can do is start this with:
```sh
CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
```
and then change the PR creation to
```sh
gh pr create -f -B $CURRENT_BRANCH -t "Release fleetd $VERSION"
```

The assumption here is that the changes would already be present on main, and backported to the patch branch.

Reply:
added this 👍

Reply:
Will the `CHANGELOG.md` be updated in `main`?

Reply:
It'll be updated on whatever branch you have checked out at the time.  For non-patch releases, that'll be `main`.

Thinking this through, if we're releasing a patch that's been backported from `main`, then the changelog would already be on the branch we're releasing, so we wouldn't have to create a new PR at all.  We'd only do this if it was a fix that was very specific to the minor version we're patching, in which case you'd be running this command while you had an rc-minor branch checked out, and the `CHANGELOG.md` would be updated in that rc-minor branch.

Reply:
What about running the PR to update the `CHANGELOG.md` based off of `main`?
(What's on `main` is the `orbit/changes/`, not the `CHANGELOG.md`, the PR is just to remove those `orbit/changes/` and move them to the `CHANGELOG.md`).

Reply:
> Right, but if we're backporting or cherry-picking a fix from main onto a prior release, the orbit/changes would be pulled into the patch branch too, no?

Correct.

I think using `main` here makes sense. Let's ship as is and we can test by releasing :)

Reply:
😬 I had already updated the code before your last comment.  It'll make the PR against main if you're pushing a tag to `main`, but if you're using a different branch it'll do a PR to that.  I don't think we want to do a patch release and then do a PR back to `main`, because `main` will either already have the changelog updates (in which case we don't need to do a PR at all), or the changes would be specific to the patch and wouldn't be correct for `main`.

If I'm missing the plot here I can change it back to `main`.